deeposlandia/postprocess.py

```diff
@@ -0,0 +1,440 @@
+        Path of the data on the file system
+    dataset : str
+        Name of the dataset
+    tile_size : int
+        Size of preprocessed images, in pixels
+
+    Returns
+    -------
+    list
+        List of dictionnaries that describes the dataset labels
+    """
+    prepro_folder = utils.prepare_preprocessed_folder(
+        datapath, dataset, tile_size, "full"
+    )
+    if os.path.isfile(prepro_folder["testing_config"]):
+        test_config = utils.read_config(prepro_folder["testing_config"])
+    else:
+        raise ValueError(("There is no testing data with the given "
+                          "parameters. Please generate a valid dataset "
+                          "before calling the program."))
+    labels = [l for l in test_config["labels"] if l["is_evaluate"]]
+    return labels
+
+
+def get_trained_model(datapath, dataset, image_size, nb_labels):
+    """Recover model weights stored on the file system, and assign them into
+    the `model` structure
+
+    Parameters
+    ----------
+    datapath : str
+        Path of the data on the file system
+    dataset : str
+        Name of the dataset
+    image_size : int
+        Image size, in pixels (height=width)
+    nb_labels : int
+        Number of output labels
+
+    Returns
+    -------
+    keras.models.Model
+        Convolutional neural network
+    """
+    K.clear_session()
+    net = SemanticSegmentationNetwork(
+        network_name="semseg_postprocessing",
+        image_size=image_size,
+        nb_labels=nb_labels,
+        dropout=1.0,
+        architecture="unet"
+    )
+    model = Model(net.X, net.Y)
+    output_folder = utils.prepare_output_folder(
+        datapath, dataset, "semantic_segmentation"
+    )
+    checkpoint_filename = "best-model-" + str(image_size) + "-full" + ".h5"
+    checkpoint_full_path = os.path.join(output_folder, checkpoint_filename)
+    if os.path.isfile(checkpoint_full_path):
+        model.load_weights(checkpoint_full_path)
+        logger.info("Model weights have been recovered from %s"
+                    % checkpoint_full_path)
+    else:
+        logger.info(("No available trained model for this image size"
+                     " with optimized hyperparameters. The "
+                     "inference will be done on an untrained model"))
+    return model
+
+
+def assign_label_colors(predicted_labels, labels):
+    """Transform raw Keras prediction into an exploitable numpy array that
+    contains label IDs
+
+    Parameters
+    ----------
+    predicted_labels : numpy.array
+        Output of the prediction process, of shape (img_size, img_size)
+    labels : list
+        List of dictionnaries that describes the dataset labels
+
+    Returns
+    -------
+    numpy.array
+        Coloured predicted labels, of shape (img_size, img_size, 3)
+    """
+    labelled_images = np.zeros(shape=np.append(predicted_labels.shape, 3),
+                               dtype=np.uint8)
+    for i in range(len(labels)):
```

Comment:
I like this form

```python
for idx, label in enumerate(labels):
    labelled_images[predicted_labels == idx] = label["color"]
```

but it's not very important.

Reply:
That's more elegant, no problem for changing this.